src/main/kotlin/org/serenityos/jakt/folding/JaktFoldingBuilder.kt

```diff
@@ -0,0 +1,51 @@
+package org.serenityos.jakt.folding
+
+import com.intellij.lang.ASTNode
+import com.intellij.lang.folding.FoldingBuilderEx
+import com.intellij.lang.folding.FoldingDescriptor
+import com.intellij.openapi.editor.Document
+import com.intellij.openapi.editor.FoldingGroup
+import com.intellij.openapi.project.DumbAware
+import com.intellij.openapi.util.TextRange
+import com.intellij.psi.PsiElement
+import com.intellij.refactoring.suggested.endOffset
+import com.intellij.refactoring.suggested.startOffset
+import org.serenityos.jakt.psi.declaration.JaktImportMixin
+import org.serenityos.jakt.psi.findChildrenOfType
+
+class JaktFoldingBuilder : FoldingBuilderEx() {
+
+    override fun buildFoldRegions(root: PsiElement, document: Document, quick: Boolean): Array<FoldingDescriptor> {
+        val descriptors = arrayListOf<FoldingDescriptor>()
+
+        // Fold import lists
+        val imports = root.findChildrenOfType<JaktImportMixin>()
```

Comment:
I tend to prefer using the non-mixin class when possible, which looks like it should be possible throughout this file